python/ray/tune/logger.py

```diff
@@ -254,11 +256,19 @@ def __init__(self, nan_str="null", **kwargs):
     def default(self, value):
         try:
             if np.isnan(value):
-                return None
-            if np.issubdtype(value, float):
-                return float(value)
-            if np.issubdtype(value, int):
+                return self.nan_str
```

Comment:
I assume this is supposed to be `self.nan_str` and not `None`. Correct me if I'm wrong.

Reply:
Actually, I think the result would be the same whether using `None` or `"null"` (which is the default of `self.nan_str`). Is there a reason for using `self.nan_str`?

Reply:
I guess it would be better, especially if we reused _SafeFallbackEncoder somewhere else